diff --git a/pyethereum/packeter.py b/pyethereum/packeter.py
index 140dfac..0663e28 100644
--- a/pyethereum/packeter.py
+++ b/pyethereum/packeter.py
@@ -65,7 +65,7 @@ class Packeter(object):
 
     # as sent by Ethereum(++)/v0.3.11/brew/Darwin/unknown
     SYNCHRONIZATION_TOKEN = 0x22400891
-    PROTOCOL_VERSION = 0x0b
+    PROTOCOL_VERSION = 424235
     NETWORK_ID = 0
     CLIENT_ID = 'Ethereum(py)/0.0.1'
     CAPABILITIES = 0x01 + 0x02 + 0x04  # node discovery + transaction relaying
diff --git a/pyethereum/processblock.py b/pyethereum/processblock.py
index 5f4dc09..776943c 100644
--- a/pyethereum/processblock.py
+++ b/pyethereum/processblock.py
@@ -320,6 +320,10 @@ def apply_op(block,tx,msg,code,compustate):
         if len(mem) < stackargs[0] + 32:
             mem.extend([0] * (stackargs[0] + 32 - len(mem)))
         v = stackargs[1]
+        if isinstance(v, (str,unicode)):
+            print v + ": "
+            v = int(v, 16)
+            print v
         for i in range(31,-1,-1):
             mem[stackargs[0]+i] = v % 256
             v /= 256
