diff --git a/pyethereum/packeter.py b/pyethereum/packeter.py
index 09267ab..a32633a 100644
--- a/pyethereum/packeter.py
+++ b/pyethereum/packeter.py
@@ -65,7 +65,7 @@ class Packeter(object):
         dict((v, k) for k, v in disconnect_reasons_map.items())
 
     SYNCHRONIZATION_TOKEN = 0x22400891
-    PROTOCOL_VERSION = 0x0c
+    PROTOCOL_VERSION = 0x6792b
     # is the node s Unique Identifier and is the 512-bit hash that serves to
     # identify the node.
     NETWORK_ID = 0
