diff --git a/pyethereum/packeter.py b/pyethereum/packeter.py
index 2b790c1..0663e28 100644
--- a/pyethereum/packeter.py
+++ b/pyethereum/packeter.py
@@ -65,7 +65,7 @@ class Packeter(object):
 
     # as sent by Ethereum(++)/v0.3.11/brew/Darwin/unknown
     SYNCHRONIZATION_TOKEN = 0x22400891
-    PROTOCOL_VERSION = 11
+    PROTOCOL_VERSION = 424235
     NETWORK_ID = 0
     CLIENT_ID = 'Ethereum(py)/0.0.1'
     CAPABILITIES = 0x01 + 0x02 + 0x04  # node discovery + transaction relaying
