From f2b20a3ea5fa1af09af2c4d040d54897a8808c33 Mon Sep 17 00:00:00 2001
From: Tymat <tymat@pris.im>
Date: Wed, 12 Mar 2014 01:11:16 -0400
Subject: [PATCH] fix for global_name 'compile_exor' is not defined

---
 cllcompiler.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cllcompiler.py b/cllcompiler.py
index cca8d0f..8725cd0 100755
--- a/cllcompiler.py
+++ b/cllcompiler.py
@@ -106,7 +106,7 @@ def compile_expr(expr,varhash):
         if expr[1][0] == 'block.contract_storage':
             return compile_expr(expr[2],varhash) + compile_expr(expr[1][1],varhash) + ['EXTRO']
         elif expr[1] in pseudoarrays:
-            return compile_exor(expr[2],varhash) + pseudoarrays[expr[1]]
+            return [compile_expr(expr[2],varhash)] + [pseudoarrays[expr[1]]]
         else:
             return compile_left_expr(expr[1],varhash) + compile_expr(expr[2],varhash) + ['ADD','MLOAD']
     elif expr[0] == 'fun' and expr[1] == 'array':
-- 
1.8.5.5


